pkg/serverless/trace/propagation/carriers.go

```diff
@@ -220,3 +222,59 @@ func rawPayloadCarrier(rawPayload []byte) (tracer.TextMapReader, error) {
 func headersCarrier(hdrs map[string]string) (tracer.TextMapReader, error) {
 	return tracer.TextMapCarrier(hdrs), nil
 }
+
+// createTraceContextFromStepFunctionInput extracts the execution ARN, state name, and state entered time and uses them to generate Trace ID and Parent ID
+func createTraceContextFromStepFunctionInput(event events.StepFunctionPayload) (*TraceContext, error) {
+	tc := new(TraceContext)
+
+	execArn := event.Execution.ID
+	stateName := event.State.Name
+	stateEnteredTime := event.State.EnteredTime
+
+	if execArn == "" || stateName == "" || stateEnteredTime == "" {
+		return nil, errorNoStepFunctionContextFound
+	}
+
+	lowerTraceID, upperTraceID := stringToDdTraceIDs(execArn)
+	parentID := stringToDdSpanID(execArn, stateName, stateEnteredTime)
+
+	tc.TraceID = lowerTraceID
+	tc.TraceIDUpper64Hex = upperTraceID
+	tc.ParentID = parentID
+	tc.SamplingPriority = sampler.PriorityAutoKeep
+	return tc, nil
+}
+
+// stringToDdSpanID hashes the Execution ARN, state name, and state entered time to generate a 64-bit span ID
+func stringToDdSpanID(execArn string, stateName string, stateEnteredTime string) uint64 {
+	uniqueSpanString := fmt.Sprintf("%s#%s#%s", execArn, stateName, stateEnteredTime)
+	spanHash := sha256.Sum256([]byte(uniqueSpanString))
+	parentID := getPositiveUInt64(spanHash[0:8])
+	return parentID
+}
+
+// stringToDdTraceIDs hashes an Execution ARN to generate the lower and upper 64 bits of a 128-bit trace ID
+func stringToDdTraceIDs(toHash string) (uint64, string) {
+	hash := sha256.Sum256([]byte(toHash))
+	lower64 := getPositiveUInt64(hash[8:16])
+	upper64 := getHexEncodedString(getPositiveUInt64(hash[0:8]))
+	return lower64, upper64
+}
+
+// getPositiveUInt64 converts the first 8 bytes of a byte array to a positive uint64
+func getPositiveUInt64(hashBytes []byte) uint64 {
+	var result uint64
+	for i := 0; i < 8; i++ {
+		result = (result << 8) + uint64(hashBytes[i])
+	}
+	result &= ^uint64(1 << 63) // Ensure the highest bit is always 0
+	if result == 0 {
+		return 1
+	}
+	return result
+}
+
+func getHexEncodedString(toEncode uint64) string {
+	//return hex.EncodeToString(hashBytes[:8])
+	return fmt.Sprintf("%x", toEncode) //maybe?
+}
```

Comment:
It might be worth considering moving all these trace id encode decode upper lower stuff to its own package. I can imagine them to be useful outside of this package in the future.

Reply:
I had a conversation with Darcy about this. We suggested that maybe it makes sense to implement this in libDatadog in Rust at some point in the far future when that's ready, but it's got to be available in Java, Go, .Net, Python, JS, etc. etc. etc.